pkg/broker/variables.go

```diff
@@ -87,3 +91,61 @@ func (bv *BrokerVariable) ToSchema() map[string]interface{} {
 
 	return schema
 }
+
+// Apply defaults adds default values for missing broker variables.
+func ApplyDefaults(parameters map[string]interface{}, variables []BrokerVariable) {
+
+	for _, v := range variables {
+		if _, ok := parameters[v.FieldName]; !ok && v.Default != nil {
+			parameters[v.FieldName] = v.Default
+		}
+	}
+
+}
+
+// ValidateVariables validates a list of BrokerVariables adhere to their JSONSchema.
+func ValidateVariables(parameters map[string]interface{}, variables []BrokerVariable) error {
+
+	allErrors := &multierror.Error{
+		ErrorFormat:utils.SingleLineErrorFormatter,
+	}
+
+	schema := createJsonSchema(variables)
+	result, err := gojsonschema.Validate(gojsonschema.NewGoLoader(schema), gojsonschema.NewGoLoader(parameters))
+
+	if err != nil {
+		multierror.Append(allErrors, err)
+	} else if len(result.Errors()) > 0 {
```

Comment:
There's a potential NPE here. gojsonschema.Validate() has a return signature of (*Result, error), and returns nil for the result when it encounters an error while loading the schema. Calling result.Errors() in such a case will cause a NPE. I suggest returning the err from Validate() directly in your if block above to avoid this problem.

Reply:
Refactored to make this a bit cleaner. The NPE problem should be resolved.